app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1844,6 +1845,36 @@ class PluginApi {
   addSidebarSection(func) {
     addSidebarSection(func);
   }
+
+  /**
+   * EXPERIMENTAL. Do not use.
+   * Register a custom renderer for a notification type or override the
+   * renderer of an existing type. See lib/notification-items/base.js for
+   * documentation and the default renderer.
+   *
+   * ```
+   * api.registerRenderDirectorForNotificationType("your_notification_type", (NotificationItemBase) => {
+   *   return class extends NotificationItemBase {
+   *     get label() {
+   *       return "some label";
+   *     }
+   *
+   *     get description() {
+   *       return "fancy description";
+   *     }
+   *   };
+   * });
+   * ```
+   * @callback renderDirectorRegistererCallback
+   * @param {NotificationItemBase} The base class from which the returned class should inherit.
+   * @returns {NotificationItemBase} A class that inherits from NotificationItemBase.
+   *
+   * @param {string} notificationType - ID of the notification type (i.e. the key value of your notification type in the `Notification.types` enum on the server side).
+   * @param {renderDirectorRegistererCallback} func - Callback function that returns a subclass from the class it receives as its argument.
+   */
+  registerRenderDirectorForNotificationType(notificationType, func) {
```

Comment:
```suggestion
  registerNotificationType(notificationType, func) {
```

From an API perspective, I think we can exclude the concept of a `renderDirector`. That feels an like internal implementation detail which does not need to be communicated to API users.

Reply:
Maybe `registerNotificationTypeRenderer`? I worry it might be too generic as `registerNotificationType`?

Reply:
That is a good point, re-reading this again `registerNotificationType` is too generic. I think `registerNotificationTypeRenderer` sounds good.

Reply:
done!